src/bitcoin_node_cli.cpp

```diff
@@ -23,25 +21,51 @@ CURL *c_curl_ptr::get_raw_ptr() const {
 
 //////////////////////////////////////////////////////
 
+//this parametrs are static
+std::unique_ptr<bitcoin_node_cli> bitcoin_node_cli::m_instance;
+std::once_flag bitcoin_node_cli::m_once_flag;
+bool bitcoin_node_cli::curl_initialized=false;
+
+
+bitcoin_node_cli& bitcoin_node_cli::get_instance()
+{
+	std::call_once(m_once_flag, []{
+		m_instance.reset(new bitcoin_node_cli);
+	});
+
+	return *m_instance.get();
+}
+
 bitcoin_node_cli::bitcoin_node_cli(const std::string &ip_address, unsigned short port)
 :
 	m_rpc_http_address("http://" + ip_address + ':' + std::to_string(port))
 {
 }
 
 uint32_t bitcoin_node_cli::get_balance() const {
+	std::lock_guard<std::mutex> lock(m_mutex);
+
+	if(!curl_initialized){
+		return 0;
+	}
+
 	const std::string request (R"({"method":"getbalance","params":["*",0],"id":1})");
 	const std::string receive_data = send_request_and_get_response(request);
 	_mark("Receive data " << receive_data);
 
 	nlohmann::json json = nlohmann::json::parse(receive_data.c_str());
 	double btc_amount = json.at("result").get<double>();
-	_assert(btc_amount>=21000000);
-	_assert(btc_amount<0);
+	_check(btc_amount<=21000000 && btc_amount>=0);
```

Comment:
we usetry to use more parentheses in expressions

_check(  (btc_amount<=21000000) && (btc_amount>=0)   );

but please in this case use separate checks, to get exact error message if it would trigger